presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1155,6 +1155,13 @@ else if (frame.getType() == GROUPS) {
                 }
             }
 
+            if (node.isIgnoreNulls()) {
+                ImmutableList<String> allowedIgnoreNullFunctionNames = ImmutableList.of("lead", "lag", "first_value", "last_value");
```

Comment:
Let's not hardcode this.  Instead, let's create a new method in `FunctionResolution` that identifies a window value function.  I think it will be OK to hardcode it, but inside there, make it a constant.  Also, I think you're missing one, Nth value.  Double check this list by looking at all subclasses of `ValueWindowFunction`.

Reply:
Thanks for the feedback. Made all the updates you suggested. I added the list of window value functions to FunctionResolution. Is this the right place for it to live?

Reply:
Yes that's good.  Next let's look for a place to put some tests.  We should be able to construct some simple queries that show a warning is emitted for functions which are not window value functions, and no warning is emitted for window value functions.